### DISCUSSION TITLE
What window is being used?

### DISCUSSION THREAD

**Discussion by ciroque** [weather-gov/api#486] [Q&A]
I have been unable to ascertain how the service determines the window of data being returned. I am using the /gridpoints/SEW endpoint. The first record seems to be some arbitrary time. How is the data window determined?

**Comment by JJKraw** [comment]
Not quite understanding the question. Can you show a specific query, the data returned, and which part of that data you are asking about?

**Comment by ciroque** [comment]
Hi, apologies for being vague. Given this query: `https://api.weather.gov/gridpoints/SEW/119,76`, as of the time of this writing the first record I get in the `properties.temperature.values[0].validTime` path has a value of  "2021-10-25T04:00:00+00:00/PT2H". 

I am wondering about the time component of that value. In previous queries the time component of the first value has been "T18:00:00+00:00". 

If I could go back in time would I find that the previous entry had a duration that ended at this time?

**Reply by JJKraw** [reply]
Not necessarily, but I'm not sure why that would matter. It might be that the temperature forecast was previously valid for three hours (from T0300) but that's not within the current slice.

If you want to know the entire period that is being reported, that's in properties.validTimes. Looking at my own area, the current gridpoint return is for over 8 days. The entire time period should be covered for each weather parameter and there should be no gaps.

**Reply by scadergit** [reply]
The gridpoint endpoint will collapse same forecast data into duration, in this case PT2H indicates that the same forecast data is valid for 2 hours versus the default one hour, or 04:00:00 and 05:00:00, and the next record would be for 06:00:00.

**Reply by scadergit** [reply]
I might have read that question too quickly, and over-simplied the answer.

**Reply by ciroque** [reply]
Appreciate the quick response!

My goal is to show a chart of the temperature over a period of 3 to 7 seven days (configurable by user). I was hoping to get the data from midnight for the current date on through the end of the data set (**Side question**: **_is it possible to limit the number of days returned?_**).

I can certainly extrapolate values backwards, or just fill them in with 0 if the data is not available. I cannot make an argument to cache data from previous queries, however as that over-complicates my app.

Back-padding shouldn't be too difficult, but if I am back-padding 18 hours worth of data it limits the usefulness of my app. So that's where my real question lies. 

**_Is there a way to consistently determine how far off midnight the first record will be?_**

The RLE I get, and it's intriguing. That should not be too problematic to unroll.

**Reply by Rhut-virani** [reply]
is there a way to get data returned from midnight?

**Reply by dwhitemv25** [reply]
Please don't necro old threads. This one is from October 2021. Please post a new question next time.

No, there isn't a way to have the start time justify to midnight, not documented anyway.  The NDFD elements that `/gridpoints` exposes don't all have hourly resolution, so in some cases their valid times will span "local midnight" instead of beginning or ending there. If it causes problems, write code that 'splits' the periods on day boundaries.  Code libraries than can manage ISO8601 dates should be able to handle it.
